tls/s2n_server_hello.c

```diff
@@ -172,6 +167,30 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
         }
     }
 
+    return 0;
+}
+
+/* Lets the client determine whether a HelloRetryRequest is valid */
+static bool s2n_server_hello_retry_is_valid(struct s2n_connection *conn)
+{
+    bool has_versions_ext = conn->server_protocol_version > 0;
```

Comment:
Is this check intended to verify that the client hello included the supported_versions extensions? Does server_protocol_version > 0 do that?

I would think that you either need to check the actual extensions stored as part of the client hello, or check that conn->server_protocol_version >= S2N_TLS13 and assume that can only happen if the extension is present.

Reply:
I've updated this to check for >= S2N_TLS13.